build.mill

```diff
@@ -194,6 +197,62 @@ object os extends Module {
     object nohometest extends ScalaTests with OsLibTestModule
   }
 
+  object zip extends Module {
+    object jvm extends Cross[OsZipJvmModule](scalaVersions)
+    trait OsZipJvmModule extends OsLibModule
+
+    val pkg = "os.shaded_org_apache_tools_zip"
+    val zipSrc = millSourcePath / "src" / "os/shaded_org_apache_tools_zip"
+
+    def apacheAntZipOriginalSource: T[PathRef] = Task(persistent = true) {
+      if (!_root_.os.exists(Task.dest / "unzipped")) {
+        val antVersion = Deps.ant.version
+        _root_.os.unzip.stream(
+          requests.get.stream(
+            s"https://repo1.maven.org/maven2/org/apache/ant/ant/$antVersion/ant-$antVersion-sources.jar"
+          ),
+          Task.dest / "unzipped"
+        )
+      }
+
+      PathRef(Task.dest / "unzipped" / "org/apache/tools/zip")
+    }
+
+    /**
+     * Shades Apache Ant
+     * [[`org.apache.tools.zip` https://ant.apache.org/manual/api/org/apache/tools/zip/package-summary.html package]] to
+     * provide Unix file permission and symbolic link support for `os.zip` and `os.unzip`
+     *
+     * A third party dependency is needed since JDK's own
+     * [[`jdk.zipfs` https://docs.oracle.com/en/java/javase/14/docs/api/jdk.zipfs/module-summary.html]] does not support
+     * symbolic links and only supports file permissions since JDK 14.
+     *
+     * Apache Ant `org.apache.tools.zip` was chosen over Apache Commons Compress due to the former not having any
+     * third party dependency, only depending on Java core libraries while the later also depends on Apache Commons IO.
+     *
+     * To avoid classpath conflicts, the dependency is shaded and compiled from source. Only the `org.apache.tools.zip`
+     * package, not the entire Ant codebase, is needed. This only adds < 100kb to Os-Lib jar size.
+     */
+    def apacheAntZipSource: T[PathRef] = Task {
```

Comment:
Can we make this a `def generatedSources` folder? That way we won't won't need to commit all the vendored code to the repo, and can rely on the build tool re-generating it on demand as necessary

Reply:
Yes